Makefile

```diff
@@ -940,3 +941,99 @@ define github_pr_add_comment
 	$(eval JSON := {"body":"$(3)"})
 	$(call github_call_api,POST,$(1)/issues/$(2)/comments,$(JSON))
 endef
+
+#####################################
+#####################################
+# ImageSet utility targets
+.PHONY: clean-imageset
+clean-imageset:
+	rm -f $(BUILD_DIR)/*imageset*
+
+
+ifdef VERSION
+OPERATOR_IMAGE ?= $(IMAGE_REGISTRY)/$(BUILD_IMAGE):$(VERSION)
+else
+OPERATOR_IMAGE ?= $(BUILD_IMAGE):latest
+endif
+CRANE_IMAGE ?= calico/go-build:master
+
+$(BINDIR)/crane:
+	mkdir -p $(BINDIR)/
+	$(eval crane_container := $(shell docker create $(CRANE_IMAGE)))
+	docker cp $(crane_container):/usr/local/bin/crane $(BINDIR)/crane
+	docker rm $(crane_container)
+
+define imageset_header
+apiVersion: operator.tigera.io/v1
+kind: ImageSet
+metadata:
+endef
+export imageset_header
+
+ifeq ($(OLD_STYLE_PRINT_IMAGE),true)
+calico_img_filter=list | grep -e /calico/ -e tigera/operator
+enterprise_img_filter=list | grep -v -e calico
+else
+calico_img_filter=listcalico
+enterprise_img_filter=listenterprise
+endif
+
+# This gen-imageset target only creates an ImageSet for the built-in registries and cannot be used
+# to generate an ImageSet for an alternate registry.
+.PHONY: gen-imageset gen-enterprise-imageset gen-calico-imageset
+gen-imageset: gen-enterprise-imageset gen-calico-imageset
+	@cat $(BUILD_DIR)/imageset-enterprise.yaml > $(BUILD_DIR)/imageset.yaml
+	@echo "---" >> $(BUILD_DIR)/imageset.yaml
+	@cat $(BUILD_DIR)/imageset-calico.yaml >> $(BUILD_DIR)/imageset.yaml
+	@echo Imageset written to file $(BUILD_DIR)/imageset.yaml
+
+.PHONY: gen-enterprise-imageset
+gen-enterprise-imageset: $(BINDIR)/crane
```

Comment:
For me the command crashes on the operator image, any chance it could be skipped?

```
2024/08/06 11:25:15 HEAD request failed, falling back on GET: HEAD https://quay.io/v2/tigera/operator/manifests/v1.35.0-0.dev-85-g1460069d5243: unexpected status code 404 Not Found (HEAD responses have no body, use GET for details)
Error: GET https://quay.io/v2/tigera/operator/manifests/v1.35.0-0.dev-85-g1460069d5243: MANIFEST_UNKNOWN: manifest unknown; map[]
```

Reply:
What operator image did you use?
No it cannot be skipped because we need the digest from the registry to create the ImageSet.
I'm guessing maybe it is trying to use an image that hasn't been built and pushed to the registry. I could add a comment/warning that running the command with an image built from master may not work if the appropriate image has not been built by CI and pushed to the registry.

Reply:
I think a comment is enough.